Bio/Entrez/Parser.py

```diff
@@ -317,10 +317,13 @@ class DataHandlerMeta(type):
 
     def __init__(cls, *args, **kwargs):
         """Initialize the class."""
+        from Bio import Entrez
+
         try:
-            cls.directory = None  # use default directory for local cache
+            cls.directory = Entrez.local_cache  # use default directory for local cache
         except PermissionError:
-            cls._directory = None  # no local cache
+            cls._directory = Entrez.local_cache  # no local cache
```

Comment:
I'm not sure the comments still make sense now... over to @mdehoon for the final say.

Reply:
That's fair, I left it to default to "None" so I figure if it turns out there's no local cache (at either the place it's been directed to, or at None for when no value was set) that's when that comment is used. Because _directory only comes in if they couldn't find the local cache (hence "no local cache").

It could be more clear as "No local cache at Entrez.local_cache" though! I could make that change rq if we want.